### ISSUE TITLE
In Game: sprite_add_ext() causes silent crash w/ some PNG images

### ISSUE THREAD

**Issue by Binsk** [YoYoGames/GameMaker-Bugs#8404]
### Description
Some PNG images cause sprite_add_ext() to silently crash the game. If loaded in sprite_add() instead then the game does not crash but the texture appears garbled.

This is inconsistent between images but if it occurs for an image it will always occur for that image.

### Steps To Reproduce
* Have a file in the "included files" directory
* Call sprite_add_ext(), specifying the file path
* Note that the game will silently crash after a moment

The example project file includes an image file that consistently causes the crash. The image loads fine in other pieces of software and seems to be a proper PNG image.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.1100.0.674 Runtime v2024.1100.0.700

### Which operating system(s) are you seeing the problem on?
Linux Arch Linux kernel 6.11.6.1
Windows 10

### Which platform(s) are you seeing the problem on?
Ubuntu, Windows

[d1cdc27d-3708-4476-b940-46f5005938d9](https://api.gamemaker.io/api/github/downloads/d1cdc27d-3708-4476-b940-46f5005938d9)

**Comment by pkraif** [comment]
For QA: The project attached has wrong logic and uses wrong accessors for `async_load` in the Async - Image Loaded event. Please replace it with the following code when testing:

```gml
if (async_load[? "id"] == sprite)
{
	if (async_load[? "status"] >= 0)
		can_draw = true;
	else
		show_message("Something went wrong loading.");
}
```

**Comment by cameron-home** [comment]
Verified fixed on Windows in IDE v2024.1300.0.718  Runtime v2024.1300.0.730. Will verify for Ubuntu tomorrow in office.

**Comment by cameron-home** [comment]
Verified fixed on Ubuntu as well in IDE v2024.1300.0.720 Runtime v2024.1300.0.733
